### DISCUSSION TITLE
Can pyright report unreachable code from a typing point of view?

### DISCUSSION THREAD

**Discussion by diepala** [microsoft/pyright#3925] [Ideas]
Could pyright report unreachable code like the following inside the if clause?

```python
def foo(mode: Literal["A", "B"]):
    if mode == 'C':
        print('Not reachable')
```

I understand that the code can be reachable at runtime, but maybe pyright could give an info message reporting this case.

**Comment by Molkree** [comment]
You will get another diagnostic message from Pyright in this case, is it not enough? It's not enabled by default though.

"Condition will always evaluate to False since the types "Literal['A', 'B']" and "Literal['C']" have no overlap"
Pyright[[reportUnnecessaryComparison](https://github.com/microsoft/pyright/blob/main/docs/configuration.md)]

![image](https://user-images.githubusercontent.com/12860284/189453903-03aaeee9-f155-46d9-ad51-554498b9c34f.png)

**Reply by diepala** [reply]
Thanks, I did not know that option
